hnn_core/network.py

```diff
@@ -947,29 +947,47 @@ def _all_to_all_connect(self, src_cell, target_cell,
             all target_type cells.
         """
         src_gids = self.gid_ranges[_long_name(src_cell)]
-        target_gids = self.gid_ranges[_long_name(target_cell)]
+        target_range = self.gid_ranges[_long_name(target_cell)]
 
         src_start = src_gids[0]  # Necessary for unique feeds
-        for target_gid in target_gids:
-            if unique:
-                src_gids = [target_gid + src_start]
-            for src_gid in src_gids:
-                if not allow_autapses and src_gid == target_gid:
-                    continue
-                self.add_connection(
-                    src_gid, target_gid, loc, receptor,
-                    nc_dict['A_weight'], nc_dict['A_delay'], nc_dict['lamtha'])
 
-    def add_connection(self, src_gid, target_gid, loc, receptor,
+        if unique:
+            src_gids = np.array(target_range) + src_start
+            src_gids = src_gids.astype(int).tolist()
+            target_gids = [[target_gid] for target_gid in target_range]
+        else:
+            target_gids = list()
+            for src_gid in src_gids:
+                target_src_pair = list()
+                for target_gid in target_range:
+                    if not allow_autapses and src_gid == target_gid:
+                        continue
+                    target_src_pair.append(target_gid)
+                target_gids.append(target_src_pair)
+
+        self.add_connection(
+            src_gids, target_gids, loc, receptor,
+            nc_dict['A_weight'], nc_dict['A_delay'], nc_dict['lamtha'])
+
+    def add_connection(self, src_gids, target_gids, loc, receptor,
                        weight, delay, lamtha):
         """Appends connections to connectivity list
 
         Parameters
         ----------
-        src_gid : int
-            Integer identifying source cell ID.
-        target_gid : list | range | int
-            Integer or list of integers identifying target cell ID.
+        src_gids : str | int | range | list of int
+            Identifier for source cells. Passing str arguments
+            ('L2_pyramidal', 'L2_basket', 'L5_pyramidal', 'L5_basket') is
+            equivalent to passing a list of gids for the relvant cell type.
+            src-target connections depend on the input type of target_gids.
+        target_gids : str | int | range | list of int | list of list of int
```

Comment:
didn't we agree not to document the `list of list of int` for now? And we use it only internally

Reply:
Wow so much discussion, only for me to completely forget to delete it 😎